instaclone/src/App.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import './App.scss';
+import {dummyData} from './dummy-data';
+import SearchBar from './components/SearchBar/SearchBar';
+import PostContainer from './components/PostSection/PostContainer';
+import PostPage from './components/PostSection/PostPage';
+import PropTypes from 'prop-types';
+import withAuthenticate from './authentication/withAuthenticate';
+import Login from './components/Login/Login';
+
+
+
+const ComponentFromWithAuthenticate = withAuthenticate(PostPage)(Login);
+
+
+class App extends React.Component {
+  
+  state = {
+    data:[],
+    search:'',
+    filteredPosts:[],
+    isLoggedIn:false
+  }
+
+
+  componentDidMount(){
+    if(localStorage.getItem('isLoggedIn')){
+      return this.setState({
+        data:dummyData
+      })
+    }
+  }
+
+  searchFilter = (e) =>{
+    const filtered = this.state.data.filter( post => 
+
+      post.username.toLowerCase().includes(e.target.value.toLowerCase()))
+      
+      this.setState({filteredPosts: filtered});
+  }
+
+  handleSearch = (event) =>{
+    this.setState({ 
+            [event.target.name]:event.target.value 
+    });
+};
+
+
+
+
+  render(){
+  
+   
+     
+
+    
+    
+    return (
+      <div className="App">
+
+        <ComponentFromWithAuthenticate data={this.state.data} filteredPosts={this.state.filteredPosts }/>
+
+        <SearchBar 
+          searchFilter={this.searchFilter} 
+          searching={this.state.search}
+          />
+    
+          <PostContainer data={this.state.data} filteredPosts={this.state.filteredPosts}
```

Comment:
the PostContainer and SearchBar should not be here. The only thing you should be rendering here is your HOC. This is also why you are getting that searchBar across your Login page. Your PostContainer should be the first HOC in withAuthenticate